addon/components/lt-infinity.js

```diff
@@ -10,62 +8,29 @@ export default Component.extend(InViewportMixin, {
   layout,
 
   rows: null,
-  scrollBuffer: 0,
+  scrollableContent: null,
+  scrollBuffer: 50,
 
   didInsertElement() {
     this._super(...arguments);
 
     let scrollBuffer = this.get('scrollBuffer');
     let width = this.$().width();
+    let scrollableContent = this.get('scrollableContent');
 
     this.setProperties({
       viewportSpy: true,
-      viewportUseIntersectionObserver: false,
       viewportTolerance: {
         left: width,
         right: width,
         bottom: scrollBuffer,
         top: 0
-      }
+      },
+      scrollableArea: scrollableContent
     });
   },
 
-  willDestroyElement() {
-    this._super(...arguments);
-    this._cancelTimers();
-  },
-
   didEnterViewport() {
-    this._debounceScrolledToBottom();
-  },
-
-  didExitViewport() {
-    this._cancelTimers();
-  },
-
-  scheduleScrolledToBottom: observer('rows.[]', 'viewportEntered', function() {
-    if (this.get('viewportEntered')) {
-      /*
-       Continue scheduling onScrolledToBottom until no longer in viewport
-       */
-      this._scheduleScrolledToBottom();
-    }
-  }),
```

Comment:
This logic is missing and would be a breaking change.

When `onScrolledToBottom` was called and the rows that were added to the body were not enough to push `{{lt-infinity}}` out of the viewport, `{{lt-body}}` kept firing `onScrolledToBottom` repeatedly until the whole viewport was filled.

I don't really like how this has been implemented originally, since `afterRender` wouldn't only be called when rows have actually been added to the table, but every time when *anything* re-renders.

https://github.com/offirgolan/ember-light-table/blob/5dccab6a47644eef0381b30004802aa88958f176/addon/components/lt-infinity.js#L55-L57

Instead of this implicit and brittle detection of table mutations, `{{lt-infinity}}` should  communicate its visibility status to `{{lt-body}}`, which in turn would explicitly observe changes to the [`rows` array](https://github.com/offirgolan/ember-light-table/blob/5dccab6a47644eef0381b30004802aa88958f176/addon/components/lt-body.js#L308) and fire `onScrolledToBottom` `afterRender` as long as `{{lt-infinity}}` is visible.

What do you think of that?

Please don't feel any responsibility to refactor this yourself—you've already done a tremendous job with this PR and I'm super thankful for that. I can also do that, if you don't have the time. :wink:

Reply:
@buschtoens Ah you are so correct about the regression.  I agree.  How about something like this? 

https://github.com/offirgolan/ember-light-table/pull/548/commits/166f1db468d7b19ef9358febb8d8b87154f0cb20

Reply:
If this is better, I can add a few scrolling tests as well.

Reply:
I wholeheartedly agree with @buschtoens's opinion above. Especially:

> I don't really like how this has been implemented originally, since afterRender wouldn't only be called when rows have actually been added to the table, but every time when anything re-renders.

@snewcomer I like your approach as an initial attempt. as @buschtoens pointed out, `lt-infinity` should probably broadcast when its actually time for `lt-body` to fire its `onScrolledToBottom` event.

Reply:
THanks @mkay581!  That was already added here - https://github.com/offirgolan/ember-light-table/commit/166f1db468d7b19ef9358febb8d8b87154f0cb20